Doctrine/UserManager.php

```diff
@@ -35,8 +34,7 @@ class UserManager extends BaseUserManager
      *
      * @param string $class
      */
-    public function __construct(PasswordUpdaterInterface $passwordUpdater, CanonicalFieldsUpdater $canonicalFieldsUpdater, ObjectManager $om, $class)
-    {
+    public function __construct(PasswordUpdaterInterface $passwordUpdater, CanonicalFieldsUpdater $canonicalFieldsUpdater, EntityManager $om, $class) {
```

Comment:
Typehinting the EntityManager implementation is a no-go:

- it breaks support for other Doctrine projects like the MongoDB ODM, as it does not depend on the `doctrine/persistence` interface anymore
- it does not even support projects decorating the ORM thanks to `EntityManagerInterface`